splinterd/src/main.rs

```diff
@@ -84,32 +84,28 @@ fn load_toml_config(config_file_path: &str) -> Config {
             "Unable to load configuration file {}: {}",
             config_file_path, err
         );
-        Config::default()
+        PartialConfig::default()
     })
 }
 
 #[cfg(feature = "config-toml")]
-fn load_toml_config(config_file_path: &str) -> Config {
-    let mut config_builder = ConfigBuilder::new();
-
+fn load_toml_config(config_file_path: &str) -> PartialConfig {
```

Comment:
Instead of returning a default, it may be better here to return None and the calling code could then avoid using it in the merge.

Reply:
I don't think this is a necessary change for this PR, because in the case that it would return None and the `toml-config` feature is enabled, the main function still expects a `config` object, which would necessitate creating a default config in the main function rather than here.

Reply:
Ok, agreed, this probably makes more sense when we are collecting a list of PartialConfigs to merge into a Config.